src/libraries/System.IO.FileSystem/tests/FileStream/ctor_str_fm_fa_fs_buffer_fo_as.Windows.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using System.Text;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public partial class FileStream_ctor_str_fm_fa_fs_buffer_fo_as : FileStream_ctor_str_fm_fa_fs_buffer_fo_as_base
+    {
+        protected override long AllocationSize => 10;
+
+        protected override long InitialLength => 0; // Windows modifies AllocationSize, but not EndOfFile (file length)
+
+        private long GetExpectedFileLength(long allocationSize) => 0; // Windows modifies AllocationSize, but not EndOfFile (file length)
+
+        private unsafe long GetActualAllocationSize(FileStream fileStream)
+        {
+            Interop.Kernel32.FILE_STANDARD_INFO info;
+
+            Assert.True(Interop.Kernel32.GetFileInformationByHandleEx(fileStream.SafeFileHandle, Interop.Kernel32.FileStandardInfo, &info, (uint)sizeof(Interop.Kernel32.FILE_STANDARD_INFO)));
+
+            return info.AllocationSize;
+        }
+
+        [PlatformSpecific(TestPlatforms.Windows)]
```

Comment:
> I would assume the FAT32 size limitation would exist independently of the OS.

When I was implementing that I did a quick check and did not find a good way of getting the file system name on every Unix. Since this is rather an edge case, I decided not to implement it for the Unixes. However, I've tested that manually and it works as expected on Linux. macOS does not report an error that we could handle properly (this is the macOS bug that I was talking about yesterday)